flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/TaskManagerLocation.java

```diff
@@ -202,7 +214,11 @@ public static String getHostName(InetAddress inetAddress) {
 			LOG.warn("No hostname could be resolved for the IP address {}, using IP address as host name. "
 				+ "Local input split assignment (such as for HDFS files) may be impacted.", inetAddress.getHostAddress());
 		} else {
-			hostName = NetUtils.getHostnameFromFQDN(fqdnHostName);
+			if (useFullHostName) {
+				hostName = fqdnHostName;
```

Comment:
this case should most certainly not be handled in this method. Have the TaskManagerRunner call `getFqdnHostName` instead (after making it public).